components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java

```diff
@@ -349,45 +371,37 @@ private void publishDynamicEventStream(String deviceType, String tenantDomain){
             eventReciverOptions.setProperty(HTTPConstants.HTTP_HEADERS, list);
             eventReciverOptions.setProperty(HTTPConstants.CUSTOM_PROTOCOL_HANDLER
                     , new Protocol(DEFAULT_HTTP_PROTOCOL
-                            , (ProtocolSocketFactory) new SSLProtocolSocketFactory(sslContext)
-                            , Integer.parseInt(Utils.replaceSystemProperty(DAS_PORT))));
+                    , (ProtocolSocketFactory) new SSLProtocolSocketFactory(sslContext)
+                    , Integer.parseInt(Utils.replaceSystemProperty(DAS_PORT))));
 
             eventStreamAdminServiceStub._getServiceClient().setOptions(eventReciverOptions);
-
-            List<String> streamList = getStreamsList(deviceType);
             for (String streamContent:streamList) {
                 JSONParser jsonParser = new JSONParser();
                 JSONObject steamJson = (JSONObject)jsonParser.parse(streamContent);
                 String name = (String) steamJson.get("name");
                 String version = (String) steamJson.get("version");
                 String streamId = name +":"+version;
-                if(eventStreamAdminServiceStub.getStreamDefinitionAsString(streamId)==null) {
-
+                if (eventStreamAdminServiceStub.getStreamDefinitionDto(streamId) == null) {
+                    log.error("its not there");
                     eventStreamAdminServiceStub.addEventStreamDefinitionAsString(streamContent);
+                } else {
+                    log.error("its there");
```

Comment:
Make this meaningful as well. See top comment. Also add context aware details to make this useful in a log file.